src/transformers/models/opt/modeling_opt.py

```diff
@@ -212,9 +211,12 @@ def forward(
                     f"Attention mask should be of size {(bsz, 1, tgt_len, src_len)}, but is {attention_mask.size()}"
                 )
             attn_weights = attn_weights.view(bsz, self.num_heads, tgt_len, src_len) + attention_mask
-            attn_weights = attn_weights.view(bsz * self.num_heads, tgt_len, src_len)
+            attn_weights = torch.max(attn_weights, torch.tensor(torch.finfo(attn_weights.dtype).min))
+            attn_weights_before_softmax = attn_weights.view(bsz * self.num_heads, tgt_len, src_len)
 
-        attn_weights = nn.functional.softmax(attn_weights, dim=-1)
+        attn_weights = nn.functional.softmax(attn_weights_before_softmax, dim=-1, dtype=torch.float32).to(
+            attn_weights_before_softmax.dtype
+        )
```

Comment:
Not entirely sure if this is the best solution, but having `attn_weights` in fp32 has helped prevent nan issues in some other models like `GPTNeo`, for example in `GPTNeo` the entire attention op is done in fp32
cf https://github.com/huggingface/transformers/blob/d453ea61205882eb4210f8c4d2fb001e2c96872b/src/transformers/models/gpt_neo/modeling_gpt_neo.py#L183-L186.

So good for me to merge if this resolves the `nan` issue and tests pass. wdyt @patrickvonplaten

Reply:
Oh sorry missed this one ! 
All OPT models are in fp16, but might be interesting in the case people would train / use it in other dtypes. Also this behavior is not observed if you cast the model in bf16 so what you are saying makes sense. Will add a small patch to add your idea

Reply:
I propose a fix here: 4c1fb44ed22dad25610788efe860f193e8cc03ec

Reply:
That works, thank you!

one last fix with the test - see my proposal and it's good to go